### ISSUE TITLE
ASTNodeList broken

### ISSUE THREAD

**Issue by thisiscam** [inducer/islpy#21]
Hi -

I was trying to perform code generation with islpy's AST package, and noticed some problem with AstNodeList class. Specifically, `AstNode.block_get_children()` raises exception, and this is caused by its internal children list not exposed by ISL's public API. 

To reproduce an exception, use the following code:
```
import islpy
ctx = islpy.Context()
schedule_map = islpy.UnionMap.read_from_str(ctx, "[N] -> { S0[i] -> [i, 0] : 0 <= i < N; S1[i] -> [i, 1] : 0 <= i < N }")
ast_build = islpy.AstBuild.from_context(islpy.Set.read_from_str(ctx, "[N] -> { : }"))
ast = ast_build.node_from_schedule_map(schedule_map)

print(ast.to_C_str())
# Prints below code:
# for (int c0 = 0; c0 < N; c0 += 1) {
#  S0(c0);
#  S1(c0);
# }

# we have S0 and S1 in a ast_node_block, which holds "children" of type ASTNodeList
body = ast.for_get_body()
assert body.get_type() == islpy.ast_node_type.block

body.block_get_children()
# Raises exception!
```

The above code outputs:
```
for (int c0 = 0; c0 < N; c0 += 1) {
  S0(c0);
  S1(c0);
}

Traceback (most recent call last):
  File "polyppl/islpy_error.py", line 18, in <module>
    body.block_get_children()
  File "/anaconda3/envs/polyppl/lib/python3.7/site-packages/islpy-2019.1.2-py3.7-macosx-10.7-x86_64.egg/islpy/_isl.py", line 106252, in block_get_children
    _result = None if (_result == ffi.NULL or _result is None) else AstNodeList(_data=_result)
  File "/anaconda3/envs/polyppl/lib/python3.7/site-packages/islpy-2019.1.2-py3.7-macosx-10.7-x86_64.egg/islpy/_isl.py", line 74, in __init__
    self._setup(_data)
  File "/anaconda3/envs/polyppl/lib/python3.7/site-packages/islpy-2019.1.2-py3.7-macosx-10.7-x86_64.egg/islpy/_isl.py", line 81, in _setup
    self._set_ctx_data()
  File "/anaconda3/envs/polyppl/lib/python3.7/site-packages/islpy-2019.1.2-py3.7-macosx-10.7-x86_64.egg/islpy/_isl.py", line 97, in _set_ctx_data
    self._ctx_data = self._get_ctx_data()
  File "/anaconda3/envs/polyppl/lib/python3.7/site-packages/islpy-2019.1.2-py3.7-macosx-10.7-x86_64.egg/islpy/_isl.py", line 550, in _get_ctx_data
    return lib.isl_ast_node_list_get_ctx(self.data)
AttributeError: cffi library '<None>' has no function, constant or global variable named 'isl_ast_node_list_get_ctx'
Exception ignored in: <function AstNodeList.__del__ at 0x7f92f80c1a60>
Traceback (most recent call last):
  File "/anaconda3/envs/polyppl/lib/python3.7/site-packages/islpy-2019.1.2-py3.7-macosx-10.7-x86_64.egg/islpy/_isl.py", line 554, in __del__
    lib.isl_ast_node_list_free(self.data)
AttributeError: cffi library '<None>' has no function, constant or global variable named 'isl_ast_node_list_free'
```

I see that this issue is not really caused by islpy, but by ISL not exposing the list interface of ast_node_list. That is, even if islpy didn't throw any exception, I still wouldn't be inspect the children list of an ast node block... related ISL discussion: https://groups.google.com/forum/#!topic/isl-development/RcNy4ogqRso

For now, I think the easiest fix would be to disable the `block_get_children` function in islpy, and document a caveat saying that AST inspection cannot be done fully -- unless the author of this repo has any better solution?

**Comment by inducer** [comment]
Fixed in fe355c3.

**Comment by inducer** [comment]
x-ref: https://gitlab.tiker.net/inducer/islpy/merge_requests/22
